src/components/utils/ManagedTree.js

```diff
@@ -46,6 +52,28 @@ let ManagedTree = React.createClass({
     this.setState({ expanded: expanded });
   },
 
+  highlightItem(highlightItems) {
+    const expanded = this.state.expanded;
+
+    // This file is visible, so we highlight it.
+    if (expanded.has(this.props.getKey(highlightItems[0]))) {
+      this.focusItem(highlightItems[0]);
+    } else {
+      let index = highlightItems.length - 1;
+
+      // Look at folders starting from the top-level until finds a
+      // closed folder and highlights this folder
+      highlightItems.some(item => {
```

Comment:
hmm, i think `highlightItems .forEach((item, index) => {})` would be cleaner that way you'll have the index and won't need to do `index--`